### PR TITLE
Allow-user-written-suggestion

### PR DESCRIPTION
**Pull request by Melesegithub**
UX/Allow user-written suggestion. I have just recorded the original sentence id and user-provided-correction on user_interaction table.

### CODE REVIEW

File: client/src/pages/UserSuggestion.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const UserSuggestion = (probs) => {
+    const [userProvidedCorrection, setUserProvidedCorrection] = useState("");
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        if (userProvidedCorrection.trim() === "") {
+          return alert("Users correction cannot be empty.");
+        }
+        const formData = { sentence: probs.randomText, user_provided_suggestion:userProvidedCorrection };
+
+        console.log(formData);
```
Comment (sztupy) [inline client/src/pages/UserSuggestion.js:15]: make sure to remove logging from published PRs

File: client/src/pages/UserSuggestion.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          } catch (error) {
+            console.log("Error submitting form:", error);
+            document.getElementById("myForm").reset();
+
+          }
+
+      };
+
+    return (
+			<div>
```
Comment (sztupy) [inline client/src/pages/UserSuggestion.js:43]: indentation is off in this section, please clean it up

File: server/api.js [hunk 3]
```diff
@@ -52,6 +52,36 @@ router.post("/save-suggestions", async (req, res) => {
...
+		}
+	} catch (error) {
+		logger.error("%O", error);
+		res
+			.status(500)
+			.json({ message: "An error occurred while saving suggestions" });
+	}
+});
+
+router.post("/save-correction", async (req, res) => {
```
Comment (sztupy) [inline server/api.js]: why the need to have this as a separate endpoint? Can't the `/save-suggestions` endpoint be updated to allow suggestions? We could then still save the shown suggestions to the database as well, which could be useful for researchers to know what options were not taken.

Comment (Melesegithub) [reply server/api.js]: I will try to fix that thanks

File: server/api.js [hunk 4]
```diff
@@ -65,6 +64,50 @@ router.post("/save-suggestions", async (req, res) => {
 	}
 });
 
+
+router.post("/save-correction", async (req, res) => {
```
Comment (sztupy) [inline server/api.js:68]: This is still an effective duplication of the `/save-suggestion` endpoint above. You can definitely design the `/save-suggestion` endpoint to support all four use cases:

- Original sentence was correct
- One of the suggestions have been picked
- None of the suggestions are correct button was used
- A typed in suggestion was used

**Review by sztupy — CHANGES_REQUESTED** [review]
Solution should use a single endpoint for all relevant functionality

**Pull request closed without merging**